crates/arrow_util/src/pretty.rs

```diff
@@ -505,15 +505,36 @@ struct ColumnWidthSizeStats {
 impl ColumnValues {
     fn try_new_from_array(col: &dyn Array, trunc: Option<usize>) -> Result<Self, ArrowError> {
         let formatter = ArrayFormatter::try_new(col, &TABLE_FORMAT_OPTS)?;
-        let vals = (0..col.len())
-            .map(|idx| {
-                let mut s = formatter.value(idx).try_to_string()?;
-                if let Some(trunc) = trunc {
-                    truncate_or_wrap_string(&mut s, trunc);
-                }
-                Ok(s)
-            })
-            .collect::<Result<Vec<_>, ArrowError>>()?;
+        let vals = match col.data_type() {
+            DataType::Float64 => {
+                // formatting float arrays
+                let floats_array = col
+                    .as_any()
+                    .downcast_ref::<Float64Array>()
+                    .unwrap()
+                    .values();
+
+                (0..floats_array.len())
+                    .map(|idx| {
+                        let s = fmt_floats(floats_array.get(idx));
+                        Ok(s)
```

Comment:
do you need to declare `s` or can you just `Ok(fmt_floats(<...>))`

Reply:
Yes I agree, My bad. I could have just written Ok(fmt_floats(<...>)). I will change this.